hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -198,8 +206,22 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
 
       setHmsTableParameters(newMetadataLocation, tbl, metadata.properties(), removedProps, hiveEngineEnabled);
 
-      persistTable(tbl, updateHiveTable);
-      threw = false;
+      try {
+        persistTable(tbl, updateHiveTable);
+        commitStatus = CommitStatus.SUCCESS;
+      } catch (Throwable persistFailure) {
```

Comment:
When this exception occurred in prod, what sort of exception did you see? 
I would have imagine any `TException` or `MetaException` implies that we do get some error response from the Metastore, a network partition e.g would have some sort of socket exception (e.g socket closed), as the HiveClientPool would try reconnecting 3 times

Reply:
In our use case we saw it as a SocketException which thrift wraps in TTransportException which is a TException, we added "Throwable" here because we were getting paranoid. Theoretically I think TException & Interrupt are the only things that can be thrown here since I don't think thrift can throw runtime exceptions in the client code, only checked ones...

Reply:
```Caused by: java.lang.RuntimeException: org.apache.thrift.transport.TTransportException: java.net.SocketTimeoutException: Read timed out```

The runtime wrapper is from how this code wraps exceptions